battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java

```diff
@@ -26,29 +26,44 @@
     private static final int PORT_NUMBER = 8081;
 
     @Override
-    public void run() throws IOException {
+    public boolean receiveServerInitialMessage() throws IOException {
         String fromServer = in.readLine();
+        boolean firstPlayer = false;
         while (!("Provide ships").equals(fromServer)) {
             if (LOGGER.isLoggable(Level.INFO)) {
                 LOGGER.info("Server: " + fromServer);
             }
             fromServer = in.readLine();
+            if ("Game has started. 1".equals(fromServer)) {
+                firstPlayer = true;
+            }
         }
+        return firstPlayer;
     }
 
     @Override
     public void send(final String allShips) throws IOException {
         out.println(allShips);
-        enemyShips = in.readLine();
+//        enemyShips = in.readLine();
     }
 
     @Override
     public String getEnemyShips() {
-        return enemyShips;
+        try {
+            return in.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
I'd use StringUtils.EMPTY instead of null if you stick to representing ships this way.